apps/music/views/artists/view.js

```diff
@@ -8,39 +8,56 @@ var ArtistsView = View.extend(function ArtistsView() {
   this.searchResults = document.getElementById('search-results');
   this.list = document.getElementById('list');
 
-  this.searchBox.addEventListener('search', (evt) => this.search(evt.detail));
+  this.client.on('databaseChange', () => this.update());
+  this.configureSearch();
+  this.configureList();
+  this.update();
+});
+
+ArtistsView.prototype.configureList = function() {
+  var list = this.list;
+
+  // Scroll search out of view, even when
+  // there arent' enough list items to scroll.
+  this.list.scrollTop = this.searchBox.HEIGHT;
+  this.list.minScrollHeight = `calc(100% + ${this.searchBox.HEIGHT}px)`;
+
+  this.list.configure({
+    getItemImageSrc: (item) => this.getThumbnail(item.name)
+  });
+
+  // Show the view only when list has something rendered,
+  // this prevents Gecko painting unnecessarily.
+  list.addEventListener('rendered', function fn() {
```

Comment:
See comment above about adding a `once()` utility method to the base view.js module.